File: python/loop_research/src/loop_research/health.py
"""Bootstrap health contract for the research worker."""

from typing import Literal

from pydantic import BaseModel, ConfigDict


class ResearchHealth(BaseModel):
    """Machine-readable readiness state."""

    model_config = ConfigDict(frozen=True)

    component: Literal["researchd"] = "researchd"
    protocol_version: Literal["loop-engine.v1alpha1"] = "loop-engine.v1alpha1"
    status: Literal["ready"] = "ready"


def research_health() -> ResearchHealth:
    """Return the local bootstrap health state."""

    return ResearchHealth()

File: apps/providerd/src/health.ts
import { z } from "zod";

export const providerHealthSchema = z.object({
  component: z.literal("providerd"),
  protocolVersion: z.literal("loop-engine.v1alpha1"),
  status: z.literal("ready"),
});

export type ProviderHealth = z.infer<typeof providerHealthSchema>;

export function provider_health(): ProviderHealth {
  return providerHealthSchema.parse({
    component: "providerd",
    protocolVersion: "loop-engine.v1alpha1",
    status: "ready",
  });
}
